packages/turbo-repository/rust/src/lib.rs

```diff
@@ -73,13 +84,36 @@ impl Package {
         workspace_path: &AbsoluteSystemPath,
     ) -> Vec<Package> {
         let node = PackageNode::Workspace(PackageName::Other(self.name.clone()));
-        let ancestors = match graph.immediate_ancestors(&node) {
-            Some(ancestors) => ancestors,
+        let pkgs = match graph.immediate_ancestors(&node) {
+            Some(pkgs) => pkgs,
             None => return vec![],
         };
 
-        ancestors
-            .iter()
+        pkgs.iter()
+            .filter_map(|node| {
+                let info = graph.package_info(node.as_package_name())?;
+                // If we don't get a package name back, we'll just skip it.
+                let name = info.package_name()?;
+                let anchored_package_path = info.package_path();
+                let package_path = workspace_path.resolve(anchored_package_path);
+                Some(Package::new(name, workspace_path, &package_path))
+            })
+            .collect()
+    }
+
+    fn dependencies(
+        &self,
+        graph: &PackageGraph,
+        workspace_path: &AbsoluteSystemPath,
+    ) -> Vec<Package> {
+        let node = PackageNode::Workspace(PackageName::Other(self.name.clone()));
+        let pkgs = match graph.immediate_dependencies(&node) {
```

Comment:
As above, this is getting just immediate dependencies, not the full transitive closure. Might be worth a comment at least.